src/scriptrag/agents/context_query.py

```diff
@@ -0,0 +1,439 @@
+            return []
+
+    def _sql_to_spec(self, sql: str, parameters: ContextParameters) -> QuerySpec:
+        """Convert raw SQL to QuerySpec.
+
+        Args:
+            sql: SQL query string
+            parameters: Context parameters for validation
+
+        Returns:
+            QuerySpec instance
+        """
+        # Extract parameter names from SQL
+        param_pattern = re.compile(r":(\w+)")
+        param_names = param_pattern.findall(sql)
+
+        # Build parameter specifications based on available context
+        from scriptrag.query import ParamSpec
+
+        param_specs = []
+        params_dict = parameters.to_dict()
+
+        for param_name in param_names:
+            if param_name in params_dict:
+                value = params_dict[param_name]
+                # Infer type from value
+                param_type: Literal["str", "int", "float", "bool"]
+                if isinstance(value, int):
+                    param_type = "int"
+                elif isinstance(value, float):
+                    param_type = "float"
+                elif isinstance(value, bool):
+                    param_type = "bool"
+                else:
+                    param_type = "str"
+
+                param_specs.append(
+                    ParamSpec(
+                        name=param_name,
+                        type=param_type,
+                        required=True,
+                        help=f"Context parameter: {param_name}",
+                    )
+                )
+
+        return QuerySpec(
+            name="context_query",
+            description="Agent context query",
+            sql=sql,
+            params=param_specs,
+        )
+
+
+class ContextResultFormatter:
+    """Formats context query results for LLM prompts."""
+
+    @staticmethod
+    def format_as_table(
+        rows: list[dict[str, Any]],
+        max_rows: int = 10,
+    ) -> str:
+        """Format results as a simple table.
+
+        Args:
+            rows: Query result rows
+            max_rows: Maximum rows to include
+
+        Returns:
+            Formatted table string
+        """
+        if not rows:
+            return "No results found"
+
+        # Limit rows
+        display_rows = rows[:max_rows]
+        truncated = len(rows) > max_rows
+
+        # Get column names
+        columns = list(display_rows[0].keys())
+
+        # Build table
+        lines = []
+
+        # Header
+        lines.append(" | ".join(columns))
+        lines.append("-" * (sum(len(col) + 3 for col in columns) - 3))
+
+        # Rows
+        for row in display_rows:
+            values = [str(row.get(col, "")) for col in columns]
+            lines.append(" | ".join(values))
+
+        if truncated:
+            lines.append(f"... and {len(rows) - max_rows} more rows")
+
+        return "\n".join(lines)
+
+    @staticmethod
+    def format_props_history(rows: list[dict[str, Any]]) -> str:
```

Comment:
This special case needs to be removed. All query results must be formatted the same way, we cannot afford to create python code for each set of query results.